quantum/split_common/transactions.c

```diff
@@ -535,7 +576,11 @@ static bool oled_handlers_master(matrix_row_t master_matrix[], matrix_row_t slav
 }
 
 static void oled_handlers_slave(matrix_row_t master_matrix[], matrix_row_t slave_matrix[]) {
-    if (split_shmem->current_oled_state) {
+    split_shared_memory_lock();
```

Comment:
The value of having a lock around a single read of a byte; either you read the value before or after the concurrent update, but there is no possibility of getting an inconsistent view of a single byte.

Reply:
You are right, I didn't think of that. I can surely change that on the other hand I'm thinking that: if someone changes the datatype in the future that doesn't have this atomic guarantees will they remember to introduce a lock here? I would bet that they don't do that, so while redundant at this point it could stay as a precaution? But I'm not commit to keeping it by all means.

Reply:
Appreciate the concern for defensive measures, though.

Reply:
Doing a checksum on a single byte isn't going to speed things up either -- may as well just transfer the actual byte unconditionally.

Reply:
You're right, but the checksum is actually meant to prevent/detect data corruption as we usually don't use any resilient to means of communication (except for odd/even parity checks for some USART peripherals).

Reply:
Understood; but in that circumstance the request/response nature of the protocol should probably be modified such that at a certain amount of data transfer, you transmit payload+checksum in a single operation. That's a longer term discussion, at least.

Reply:
Sure, does this prevent merging the changes in this PR?

Reply:
No.